src/rect.rs

```diff
@@ -621,6 +621,11 @@ impl Shape for Rect {
     fn as_rect(&self) -> Option<Rect> {
         Some(*self)
     }
+
+    #[inline]
+    fn contains(&self, pt: Point) -> bool {
+        self.contains(pt)
```

Comment:
I'm surprised this works without complaint, because `contains` is ambiguous? I would expect to have to do `Rect::contains(pt)`.

In any case, I agree this makes sense, although it will only matter in very specific circumstances; normally a method call that exists on a trait and a concrete type will resolve to the impl on the concrete type, so this would only be used in situations where something was passed as a generic `Shape`. Still worth doing though!

Reply:
I am using generic shapes and this was a semi-major bottleneck for me. Not a real bottleneck, still waiting on my `image` improvements to land but still a good 30% or so. This will only help when I am using rects, but that will be the majority of the time.

For other types `winding` is basically `contains` so those dont have as big of a performance hit.

> I'm surprised this works without complaint

IIRC the default is always on the type itself if it exists, only when it cant find that traits are considered.